File: test-first-sample.js
#!/usr/bin/env node

const {PRNG} = require('./../pokemon-showdown-ts/dist/sim/prng');
const Dex = require('./../pokemon-showdown-ts/dist/sim/dex').Dex;

const dex = Dex.mod('gen9');
const prng = new PRNG([0, 0, 0, 1]);

let allSpecies = Object.values(dex.species.all());
allSpecies.sort((a, b) => a.name.localeCompare(b.name));

console.log('Total species:', allSpecies.length);
console.log('\n=== First PRNG call (for species selection) ===');
const rawValue = prng.random();  // Get raw [0,1) value
console.log('Raw random value:', rawValue);

const index = Math.floor(rawValue * allSpecies.length);
console.log('Calculated index:', index);
console.log('Species at that index:', allSpecies[index].name);

console.log('\nSpecies around that index:');
for (let i = Math.max(0, index - 2); i <= Math.min(allSpecies.length - 1, index + 2); i++) {
    console.log(`  [${i}]: ${allSpecies[i].name}${i === index ? ' <-- selected' : ''}`);
}

File: examples/test_first_sample.rs
/// Trace first species selection in Rust

use pokemon_showdown::{Dex, PRNG, PRNGSeed};

fn main() {
    let dex = Dex::load_default().expect("Failed to load dex");
    let mut prng = PRNG::new(Some(PRNGSeed::Gen5([0, 0, 0, 1])));

    let mut all_species: Vec<_> = dex.species.values().collect();
    all_species.sort_by_key(|s| &s.name);

    println!("Total species: {}", all_species.len());
    println!("\n=== First PRNG call (for species selection) ===");

    let index = prng.random_range(0, all_species.len() as i32) as usize;
    println!("Calculated index: {}", index);
    println!("Species at that index: {}", all_species[index].name);

    println!("\nSpecies around that index:");
    let start = if index >= 2 { index - 2 } else { 0 };
    let end = (index + 3).min(all_species.len());
    for i in start..end {
        println!("  [{}]: {}{}", i, all_species[i].name, if i == index { " <-- selected" } else { "" });
    }
}
